File: Difficulty: Easy/Balanced Tree Check/balanced-tree-check.cpp
/* A binary tree node structure

class Node {
  public:
    int data;
    Node* left;
    Node* right;

    // Constructor to initialize a new node
    Node(int val) {
        data = val;
        left = NULL;
        right = NULL;
    }
};
 */

class Solution {
  public:
    int dfs(Node* root, int level, int& ans) {
        if(root==nullptr) return 0;
        int left = dfs(root->left,level+1,ans);
        if(left==-1) return -1;
        int right = dfs(root->right,level+1,ans);
        if(right==-1) return -1;
        if(abs(left-right)>1) {
            ans = -1;
            return -1;
        }
        return 1+max(left,right);
    }
    bool isBalanced(Node* root) {
        // Code here
        int level = 0;
        int ans = 0;
        dfs(root,level,ans);
        if(ans == -1) return false;
        return true;
    }
};

File: Difficulty: Easy/Balanced Tree Check/balanced-tree-check.java
/* Structure of binary tree node
class Node {
    int data;
    Node left, right;

    Node(int d)
    {
        data = d;
        left = right = null;
    }
}*/

class Solution {
    private int solve(Node root) {
        if(root==null) return 0;
        int left = solve(root.left);
        if(left == Integer.MIN_VALUE) return left;
        int right = solve(root.right);
        if(right == Integer.MIN_VALUE) return right;
        
        if(Math.abs(left-right)>1) return Integer.MIN_VALUE;
        
        return 1+Math.max(left,right);
    }
    public boolean isBalanced(Node root) {
        // code here
        
        int ans = solve(root);
        if(ans!=Integer.MIN_VALUE) return true;
        return false;
        
    }
}